stdlib/public/SDK/Foundation/FileManagerThunks.m

```diff
@@ -17,19 +17,19 @@
     NSFileManager *NS_RELEASES_ARGUMENT _Nonnull self_,
     NSURL *NS_RELEASES_ARGUMENT _Nonnull originalItemURL,
     NSURL *NS_RELEASES_ARGUMENT _Nonnull newItemURL,
-    NSString *NS_RELEASES_ARGUMENT _Nonnull backupItemName, NSUInteger options,
+    NSString *NS_RELEASES_ARGUMENT _Nullable backupItemName, NSUInteger options,
     NSError *_Nullable *_Nullable error) {
 
   NSURL *result = nil;
-  [self_ replaceItemAtURL:originalItemURL
-            withItemAtURL:originalItemURL
-           backupItemName:backupItemName
-                  options:(NSFileManagerItemReplacementOptions)options
-         resultingItemURL:&result
-                    error:error];
+  BOOL success = [self_ replaceItemAtURL:originalItemURL
+                           withItemAtURL:newItemURL
+                          backupItemName:backupItemName
+                                 options:(NSFileManagerItemReplacementOptions)options
+                        resultingItemURL:&result
+                                   error:error];
   [self_ release];
   [originalItemURL release];
   [newItemURL release];
   [backupItemName release];
-  return [result retain];
+  return success ? [result retain] : nil;
```

Comment:
It doesn't leave garbage to my knowledge in success or failure, but while I was in here, I simply made sure we use "proper" ObjC return-by-ref coding practices. I don't expect this particular change would have any noticeable effect on the behavior of the function.